airflow/api_fastapi/core_api/serializers/dag_run.py

```diff
@@ -18,13 +18,28 @@
 from __future__ import annotations
 
 from datetime import datetime
+from enum import Enum
 
 from pydantic import BaseModel, Field
 
 from airflow.utils.state import DagRunState
 from airflow.utils.types import DagRunTriggeredByType, DagRunType
 
 
+class DAGRunModifyStates(str, Enum):
+    """Enum for DAG Run states when updating a DAG Run."""
+
+    QUEUED = DagRunState.QUEUED
+    SUCCESS = DagRunState.SUCCESS
+    FAILED = DagRunState.FAILED
+
+
+class DAGRunPatchBody(BaseModel):
+    """DAG Run Serializer for PATCH requests."""
+
```

Comment:
Is it appropriate to add the dag run note here too?

I know its a different model, but for the user that doesn't matter. I definitely see value in adding a note when you manually change the run's state.

Reply:
We also have another endpoint to update dag run note specifically - [Link](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/set_dag_run_note)

But, I think it does make sense to give the ability to update the dag run note as well. Maybe we should consider as a feature request to AF3?

Reply:
Yes we can definitely make the patch broader than just the `state`.

Maybe we can merge this one as is, and @rawwar can follow up with a feature PR to make transform this route into a more general `patch dag_run` that allow update on both `state` and `note content` ?

Reply:
And I think dedicated endpoint to update the note can just be removed (not migrated at all and deleted) if the `Patch` handles that.

Reply:
Happy for it to be a follow up PR. But if we're rewriting the API. then let's also improve it!

Reply:
Thanks, resolving so I can merge